push.sh

```diff
@@ -1,16 +1,72 @@
 #! /bin/sh
 
-# Use WEBDAV_USERNAME or WEBDAV_PASSWORD as default, if provided
+# Use WEBDAV_USERNAME as default, if provided.
 if [ -z "$PLUGIN_USERNAME" ] && [ ! -z "$WEBDAV_USERNAME" ]; then
+
     PLUGIN_USERNAME="$WEBDAV_USERNAME"
 fi
+
+# Use WEBDAV_PASSWORD as default, if provided.
 if [ -z "$PLUGIN_PASSWORD" ] && [ ! -z "$WEBDAV_PASSWORD" ]; then
+
     PLUGIN_PASSWORD="$WEBDAV_PASSWORD"
 fi
 
 # If username and password are provided, add auth
 if [ ! -z "$PLUGIN_USERNAME" ] && [ ! -z "$PLUGIN_PASSWORD" ]; then
-    AUTH="-u ${PLUGIN_USERNAME}:${PLUGIN_PASSWORD}"
+
+    AUTH="--user ${PLUGIN_USERNAME}:${PLUGIN_PASSWORD}"
+fi
+
+# Use a proxy, if one is specified
+if [ ! -z "$PLUGIN_PROXY_URL" ]; then
+
+    PLUGIN_PROXY_URL="--proxy ${PLUGIN_PROXY_URL}"
 fi
 
-curl -T $PLUGIN_FILE $AUTH $PLUGIN_DESTINATION
+# If a timeout is specified, make use of it. If none is defined, use a default value (30 seconds).
+if [ ! -z "$PLUGIN_TIMEOUT" ]; then
+
+    PLUGIN_TIMEOUT="--max-time ${PLUGIN_TIMEOUT}"
+else
+
+    PLUGIN_TIMEOUT="--max-time 1800"
+fi
+
+# Set PLUGIN_ATTEMPTS to one if nothing else is specified
+if [ -z "$PLUGIN_ATTEMPTS" ]; then
+
+    PLUGIN_ATTEMPTS=1
+fi
+
+# Repeat the upload as long as specified.
+while [ $PLUGIN_ATTEMPTS -gt 0 ]; do
+
+  # Uploading the file
+  curl $PLUGIN_PROXY_URL --upload-file $PLUGIN_FILE $AUTH $PLUGIN_DESTINATION --progress-bar $PLUGIN_TIMEOUT
```

Comment:
Also, is it necessary for the progress bar and timeout flags to go after the destination? It would be preferable to have `<command> <opts> <args>` rather than a mix.